### DISCUSSION TITLE
Creazione fattura: http error 409

### DISCUSSION THREAD

**Discussion by mesposito-bytecode** [fattureincloud/api#50] [Q&A] [answered]
Salve,

abbiamo un'applicazione che si integra con le vostre API per generare fatture.
Da qualche giorno nei casi di richieste parallele o piuttosto vicine di generazione di fatture, stiamo ricevendo i seguenti errori.
`Error to create invoice: 409 Conflict: Esiste già un documento con lo stesso numero.` 

Le nostre richieste, settano nel body solamente il campo "numeration", ma non il campo "number", in modo tale che "l'autoincrement" venga gestito dall'api di creazione.

Grazie del supporto

**Comment by emajo** [comment, marked as answer]
Salve @mesposito-bytecode, putroppo non gestiamo la creazione parallela di fatture, l'errore è dato dal fatto che, se due fatture vengono create contemporaneamente gli verrà assegnato lo stesso number, perciò la fattura che verrà salvata per prima non riscontrerà problemi, mentre la seconda darà appunto `409 conflict`, le suggerisco di gestire sincronamente la creazione delle risorse, in caso non fosse possibile potrebbe implementare un meccanismo per assegnare un number sempre diverso per far si che non vadano in conflitto.

Grazie,
Emanuele
